### PR TITLE
Rename defaultClientConfiguration to defaultExtensionConfiguration

### PR DESCRIPTION
**Pull request by AndrewFossAWS**
#
* See smithy-typescript changes: https://github.com/awslabs/smithy-typescript/pull/889
* See client changes in aws-sdk-js-v3: https://github.com/aws/aws-sdk-js-v3/pull/5120

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/smithy-client/src/extensions/defaultClientConfiguration.ts [hunk 1]
```diff
@@ -1,13 +1,8 @@
...
-
-/**
- * @internal
- */
-export type DefaultExtensionConfigType = Parameters<typeof getChecksumConfiguration>[0];
+import { getChecksumConfiguration } from "./checksum";
+import { DefaultExtensionConfigType } from "./defaultExtensionConfiguration";
 
 /**
+ * @deprecated will be replaced by getDefaultExtensionConfiguration
```
Comment (kuhe) [inline packages/smithy-client/src/extensions/defaultClientConfiguration.ts:5]: move this as a stub into defaultExtensionConfiguration?

```ts
export const getDefaultClientConfiguration = getDefaultExtensionConfiguration;
```

Comment (kuhe) [reply packages/smithy-client/src/extensions/defaultClientConfiguration.ts:5]: that way this file can be deleted. We maintain package API compatibility at the index, but not down to the filename level.

File: packages/smithy-client/src/extensions/checksum.ts [hunk 2]
```diff
@@ -1,37 +1,5 @@
...
-  algorithmId(): AlgorithmId;
-  checksumConstructor(): ChecksumConstructor | HashConstructor;
-}
-
-/**
- * @internal
- *
- * @deprecated will be imported from types.
- */
-export interface ChecksumConfiguration {
```
Comment (kuhe) [inline packages/smithy-client/src/extensions/checksum.ts:31]: maintain a re-export of these for API compatibility

```ts
export { AlgorithmId, ChecksumAlgorithm, ChecksumConfiguration };
```

**Comment by kuhe** [comment]
you would have to commit and publish the TS packages, and only afterward commit the java codegen and JSv3 codegen changes

**Review by kuhe — APPROVED** [review]

File: packages/smithy-client/src/extensions/defaultExtensionConfiguration.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    ...getChecksumConfiguration(runtimeConfig),
+  };
+};
+
+/**
+ * @deprecated use getDefaultExtensionConfiguration
+ *
+ * Helper function to resolve default extension configuration from runtime config
+ */
+export const getDefaultClientConfiguration = getDefaultExtensionConfiguration;
```
Comment (syall) [inline packages/smithy-client/src/extensions/defaultExtensionConfiguration.ts:26]: nit: should this still also be marked as `@internal`?

**Review by syall — APPROVED** [review]

**Pull request merged**